paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -1145,8 +1148,28 @@ public final void createPartition(
             CatalogPartitionSpec partitionSpec,
             CatalogPartition partition,
             boolean ignoreIfExists)
-            throws CatalogException {
-        throw new UnsupportedOperationException();
+            throws CatalogException, PartitionAlreadyExistsException {
+        if (partitionExists(tablePath, partitionSpec)) {
+            if (!ignoreIfExists) {
+                throw new PartitionAlreadyExistsException(getName(), tablePath, partitionSpec);
+            }
+        }
+        try {
+            if (catalog instanceof CachingCatalog) {
```

Comment:
Why need to check CachingCatalog?

Reply:
<img width="1036" alt="image" src="https://github.com/user-attachments/assets/7b8013cc-6373-4a39-a314-8249aa2fab4d">
Here catalog is CachingCatalog which wraped HiveCatalog, so handle it as CacheingCatalog firstly. @JingsongLi